ivadomed/loader/utils.py

```diff
@@ -66,9 +66,16 @@ def split_dataset(df, center_test_lst, split_method, random_seed, train_frac=0.8
 
     elif split_method == 'per_patient':
         # Separate dataset in test, train and validation using sklearn function
-        X_train, X_remain = train_test_split(df['participant_id'].tolist(), train_size=train_frac,
-                                             random_state=random_seed)
-        X_test, X_val = train_test_split(X_remain, train_size=test_frac / (1 - train_frac), random_state=random_seed)
+        if test_frac == 1:
+            X_test = df['participant_id'].tolist()
+        else:
+            X_train, X_remain = train_test_split(df['participant_id'].tolist(), train_size=train_frac,
+                                                 random_state=random_seed)
+            if test_frac == 0:
+                X_val = X_remain
+            else:
+                X_test, X_val = train_test_split(X_remain, train_size=test_frac / (1 - train_frac),
+                                                 random_state=random_seed)
```

Comment:
So.. if I understand well:
- you will have two distinct datasets: one for training / validation, another one for testing
- at training time, you will put `test_frac=0`
- at testing tome, `test_frac=1`

Another solution.. would be to generate a `joblib` file containing the customed split and use it with `fname_split`. So that no need to modify this param between the training / testing... dunno what's easier.

Reply:
Update: Mmmm I ve just seen the changes in `main.py`... Will think further about this split

Reply:
Yes I agree the workflow could be re-thought, not super convenient

Reply:
The case where we would want to use "segment" rather than "test" is probably only when we have unlabeled datasets (in a different BIDS folder than the labeled dataset). Maybe when "segment" is chosen, the whole dataset can be segmented, this way we avoid having to change test_frac at training and testing time.

Reply:
> Maybe when "segment" is chosen, the whole dataset can be segmented, this way we avoid having to change test_frac at training and testing time.

Agree!